### ISSUE TITLE
[a11y]: Tearsheet Menu guidance causes developers to use inaccessible component

### ISSUE THREAD

**Issue by samueldmeyer** [carbon-design-system/ibm-products#8833]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

Windows, MacOS

### Package version

v2.80.0

### Automated testing tool and ruleset

Manual testing with VoiceOver and JAWS

### Assistive technology

JAWS, VoiceOver

### Description

Currently, the [Tearsheet guidance](https://pages.github.ibm.com/carbon/ibm-products/components/tearsheet/style/#4.-influencers) for an Influencer Menu (under "Combinations, spacing & alignment") suggests using the in-product navigation from the UI Shell which is implemented as a [SideNavMenuItem](https://react.carbondesignsystem.com/?path=/docs/components-ui-shell-sidenav--overview) which renders a link. However, designers who want a Menu on a Tearsheet intend it to change only the content of the Tearsheet, so developers should use a button, not a link. As a result, developers end up implementing a link with a click event, which either results in the menu item not being keyboard accessible (if no href is added) or read as a "link" by screen readers.

Developers need an example of the Influencer Menu pattern in the Storybook that correctly uses buttons. This may require adding a "SideNavButton" that is similar to the "SideNavLink" but is meant to be used as a button.

### WCAG 2.1 Violation

Success Criterion 4.1.2 Name, Role, Value (Level A)

### Reproduction/example

https://stackblitz.com/edit/github-yteyauuf?file=src%2FApp.tsx

### Steps to reproduce

Use a screen reader and move the cursor to to the menu items.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by amal-k-joy** [comment]
Hi @samueldmeyer ,
I was wondering if you can pass the prop `as = "button"` for [SideNavMenuItem](https://react.carbondesignsystem.com/?path=/docs/components-ui-shell-sidenav--overview) that will render button or any element as SideNavMenuItem

**Comment by samueldmeyer** [comment]
@amal-k-joy If I pass `as = "button"` then it loses all styling, which sort of defeats the purpose of using the component. I updated the stackblitz with an example of attempting that.

**Comment by amal-k-joy** [comment]
> [@amal-k-joy](https://github.com/amal-k-joy) If I pass `as = "button"` then it loses all styling, which sort of defeats the purpose of using the component. I updated the stackblitz with an example of attempting that.

@samueldmeyer , Just to confirm — my understanding is that you're looking for an example where it visually renders like a side nav item, but internally renders a button, correct?

If so, this behavior seems to originate from the Carbon SideNav component. It might be best to raise an issue on their side requesting support for SideNav styling when the as prop is used to render a different underlying element.

**Comment by samueldmeyer** [comment]
For reference, here is the existing request for SideNavMenuItem to be able to render as a button: https://github.com/carbon-design-system/carbon/issues/18562

**Comment by samueldmeyer** [comment]
This remains an open request, blocked by https://github.com/carbon-design-system/carbon/issues/18562. How do we keep it open and avoid it being closed due to being stale?

**Comment by samueldmeyer** [comment]
not stale

**Comment by samueldmeyer** [comment]
@amal-k-joy can you fix the labels so this will stop being flagged as stale?

**Comment by amal-k-joy** [comment]
Hi @samueldmeyer,
Do you think we can close this issue for now? It looks like https://github.com/carbon-design-system/carbon/issues/18562 should address it, and I don’t see any changes needed from the Tearsheet component at this point.

**Comment by samueldmeyer** [comment]
This request is for an example in the Tearsheet Storybook. It is blocked by https://github.com/carbon-design-system/carbon/issues/18562. If you have decided you won't ever add that example, you can close this. But I would prefer that you eventually add an example once it is possible!
